src/p_map.c

```diff
@@ -1883,7 +1883,7 @@ static boolean PIT_ChangeSector(mobj_t *thing)
       P_SetMobjState(thing, S_GIBS);
       thing->flags &= ~MF_SOLID;
       thing->height = thing->radius = 0;
-      if (STRICTMODE(colored_blood))
+      if (NOTSTRICTMODE(colored_blood))
```

Comment:
Do we even need to check `strictmode` anymore?

Reply:
I think so, because in `strictmode` we always have to show colored blood for the PWADs that set it.
Alternatively, we can make our `colored_blood` setting only for vanilla IWAD monsters.

Reply:
Hm, this is a similar question to applying translucency to sprites. Does disabling the feature mean to disable it even for PWADs that enable it for certain sprites? Currently, we do so. Should PWAD choices override user choices?

Reply:
> Should PWAD choices override user choices?

Yes, I think so. However, PrBoom+ does not provide these options.

Reply:
Yes, but PrBoom+ allows to disable translucency only for predefined things.

Reply:
So what should we do? I like to leave options for vanilla IWAD things only. I remember there was MBF21 Heretic in Doom PWAD with translucent monsters, so turning off translucency for it seems wrong.

Reply:
Yes, I agree!

Reply:
Regarding translucency, since there is not a single translucent wall in any IWAD, the `general_translucency & TRANSLUCENCY_WALLS` case should probably always be enabled?

Reply:
> the `general_translucency & TRANSLUCENCY_WALLS` case should probably always be enabled?

Yes, I think so. So the only option left is translucency for IWAD things.

Reply:
Yep!

Reply:
> Hm, this is a similar question to applying translucency to sprites. Does disabling the feature mean to disable it even for PWADs that enable it for certain sprites? Currently, we do so. Should PWAD choices override user choices?

We will need something similar to this to accomplish that for translucency:
https://github.com/coelckers/prboom-plus/blob/d7f95371d59e6889c61f02aea65e2d28a9f2b2a5/prboom2/src/d_deh.c#L1506

Reply:
PrBoom+ doesn't seem to care if PWAD enables or disables translucency for vanilla things. Should we move the initialization of all tables from `d_main.c` to `d_deh.c`?

Reply:
> PrBoom+ doesn't seem to care if PWAD enables or disables translucency for vanilla things. 

There is always room for improvement. I like the logic you applied to our colored blood implementation.

> Should we move the initialization of all tables from `d_main.c` to `d_deh.c`?

If it helps? What what change?

Reply:
There is an important difference between translucency and colored blood: Boom compatible ports always had a number of map objects with predefined translucency, whereas nothing like this existed for colored blood before. This means, if a DEHACKED lump in a PWAD applies the translucency flag for one map object, it does not necessarily mean that all other map objects should *not* have translucency.

Reply:
So, we only turn off translucency for things that are predefined in MBF, and if PWAD changes any of them, we disable this option. Alternatively, we can disable or enable transparency for all predefined things, regardless of whether PWAD changes them, but we will not change new things, as in PrBoom+.

In practice, people want to turn off translucency for fireballs, as it's not vanilla and sometimes changes gameplay a lot (eg Arch-Ville fire).

Reply:
> So, we only turn off translucency for things that are predefined in MBF, and if PWAD changes any of them, we disable this option. Alternatively, we can disable or enable translucency for all predefined things, regardless of whether PWAD changes them, but we will not change new things, as in PrBoom+.

I don't know what's right or wrong, I always play with translucency enabled. All I want is a switch that does what people expect and respect PWAD authors' choices. The best thing to do is probably to act as PrBoom+ if the "general translucency" option was enabled and turn our menu switch into PrBoom+'s "translucency for predefined things" pendant. Translucency for walls should be unaffected by this and always be enabled, but I think we already agreed upon this.